cds-feature-attachments/src/main/java/com/sap/cds/feature/attachments/handler/applicationservice/helper/ModifyApplicationHandlerHelper.java

```diff
@@ -45,11 +45,10 @@ public static InputStream handleAttachmentForEntity(List<CdsData> existingDataLi
 		var keys = ApplicationHandlerHelper.removeDraftKeys(path.target().keys());
 		ReadonlyDataContextEnhancer.fillReadonlyInContext((CdsData) path.target().values());
 		var existingData = getExistingData(keys, existingDataList);
-		var contentIdExists = path.target().values().containsKey(Attachments.CONTENT_ID);
 		var contentId = (String) path.target().values().get(Attachments.CONTENT_ID);
 
 		// for the current request find the event to process
-		ModifyAttachmentEvent eventToProcess = eventFactory.getEvent(content, contentId, contentIdExists, existingData);
```

Comment:
@adamandre79 Hi Andre, I think the boolean parameter `contentIdExists` can be removed from the signature of method `eventFactory.getEvent`. From my point of view it can be replaced with a null-check of the parameter `contentId` inside the getEvent method.

But I wonder about this test case: https://github.com/cap-java/cds-feature-attachments/blob/7a92bfcb0679430b9fb3f4c2b8c905d94d422fe7/cds-feature-attachments/src/test/java/com/sap/cds/feature/attachments/handler/applicationservice/processor/modifyevents/DefaultModifyAttachmentEventFactoryTest.java#L131

This tests calls this method with parameter combinations that will not occur during runtime. Can I remove this test ? Is it really a valid test-case ? Do you remember why it was implemented this way ?

Thanks,
Markus

Reply:
yes the variable can be replaced, I did not wanted to check for null inside, but this is personal preference.
The test was implemented to make the method more robust, it can also be removed

Reply:
with removing the parameter `contentIdExists` I was also able to remove some code, which was never executed anymore. It is no longer possible to call the method `getEvent` with an inconsistency between the parameters `contentId` and `contentIdExists`, because the null-check is done inside the method.
This was also the cause for the dropped code coverage.